primer-miso/primer-miso.cabal

```diff
@@ -0,0 +1,107 @@
+cabal-version: 3.0
+name:          primer-miso
+version:       0.8.0.0
+license:       AGPL-3.0-or-later
+license-file:  COPYING
+copyright:     (c) 2024 Hackworth Ltd
+maintainer:    src@hackworthltd.com
+author:        Hackworth Ltd <src@hackworthltd.com>
+stability:     experimental
+synopsis:
+  A web frontend for building Primer programs, using the Miso framework
+
+category:      UI
+
+library
+  exposed-modules:
+    Primer.Miso
+    Primer.Miso.Colors
+    Primer.Miso.Layout
+    Primer.Miso.Util
+
+  hs-source-dirs:     src
+  default-language:   GHC2021
+  default-extensions:
+    DataKinds
+    DeriveAnyClass
+    DerivingStrategies
+    DerivingVia
+    LambdaCase
+    NoImplicitPrelude
+    OverloadedStrings
+
+  ghc-options:
+    -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates
+    -Wcompat -Widentities -Wredundant-constraints
+    -Wmissing-deriving-strategies -fhide-source-paths
+
+  build-depends:
+    , aeson           >=2.0      && <2.3
+    , base            >=4.12     && <4.21
+    , bytestring      >=0.10.8.2 && <0.13
+    , containers      >=0.6.0.1  && <0.7.0
+    , data-default    ^>=0.8.0.0
+    , deriving-aeson  >=0.2      && <0.3.0
+    , extra           >=1.7.10   && <1.8.0
+    , jsaddle         ^>=0.9.9.2
+    , linear          ^>=1.23
+    , miso            ^>=1.8.5.0
+    , mtl             >=2.2.2    && <2.4.0
+    , optics          >=0.4      && <0.5.0
+    , primer          ^>=0.7.2
+    , text            >=2.0      && <2.2
+    , uniplate        >=1.6      && <1.7.0
+
+  if arch(wasm32)
+    build-depends: jsaddle-wasm ^>=0.0.1.0
+
+  else
+    build-depends:
+      , jsaddle-warp  ^>=0.9.9.2
+      , warp          >=3.3       && <3.5
+      , websockets    ^>=0.13.0.0
+
+executable primer-miso
+  main-is:            Main.hs
+  hs-source-dirs:     exe
+  default-language:   GHC2021
+  default-extensions:
+    DataKinds
+    DeriveAnyClass
+    DerivingStrategies
+    DerivingVia
+    LambdaCase
+    NoImplicitPrelude
+    OverloadedStrings
+
+  ghc-options:
+    -Wall -Wincomplete-uni-patterns -Wincomplete-record-updates
+    -Wcompat -Widentities -Wredundant-constraints
+    -Wmissing-deriving-strategies -fhide-source-paths
+
+  build-depends:
+    , base
+    , jsaddle
+    , primer
+    , primer-miso
+
+  if arch(wasm32)
+    build-depends:
+      , ghc-experimental  ^>=0.1.0.0
+      , jsaddle-wasm
+
+    ghc-options:
+      -no-hs-main -optl-mexec-model=reactor "-optl-Wl,--export=hs_start"
+
+  else
+    build-depends:
+      , jsaddle-warp
+      , warp
+      , websockets
+
+    ghc-options:   -threaded -rtsopts -with-rtsopts=-N
+
+  extra-source-files:
+    build-frontend.sh
+    frontend/**/*.html
+    frontend/**/*.js
```

Comment:
`Warning: primer-miso.cabal:104:3: Unknown field: "extra-source-files"`

I don't think this does anything in this position.

Reply:
Yes, I have a fix for this.

Reply:
Fixed.